### PR TITLE
Refactor tests

### PR DESCRIPTION
**Pull request by RageCage64**
Made some adjustments to the HttpHelpers that Shep wrote. Changed it to a client wrapper around an http client with extension methods. Should make it slightly more extensible and reduce boilerplate in actual tests. Move all tests over to using the new client wrapper.

Also flattened the Models/Requests folder, since it didn't add much for each resource to have its own folder. We may go back to that in the future, but for now its easier for it just to be flat.

### CODE REVIEW

File: LeaderboardBackend/Models/Requests/ParticipationRequests.cs [hunk 1]
```diff
@@ -18,3 +18,10 @@ public class CreateParticipationRequest
 	public bool IsSubmitter { get; set; } = true;
 }
 
+public class UpdateParticipationRequest
+{
+	public string Comment { get; set; } = "";
+
+	[Required(ErrorMessage = "Please add a VoD link. Your participation can't be confirmed otherwise.")]
```
Comment (zysim) [inline LeaderboardBackend/Models/Requests/ParticipationRequests.cs:25]: Maybe we should make a custom rule for this such that it's both required and that it has to be a link to a valid video, or a link from a set list of domains.

**Review by zysim — COMMENTED** [review]
Apart from everything, I'd also like to suggest if we could name Requests as `Models/Request/<ModelName>.cs`, instead of `<ModelName>Request.cs`. What do you think?

File: LeaderboardBackend/Models/Requests/ParticipationRequests.cs (hunk 1 above)
Comment (zysim) [reply LeaderboardBackend/Models/Requests/ParticipationRequests.cs:25]: We could just add a FIXME on this, for a future PR to tackle

**Comment by RageCage64** [comment]
> Apart from everything, I'd also like to suggest if we could name Requests as Models/Request/<ModelName>.cs, instead of <ModelName>Request.cs. What do you think?

Would prefer not, it's really annoying having lots of files with the same name open in the editor. I can change it if you don't think that's a big deal though.

File: LeaderboardBackend.Test/Modships.cs [hunk 2]
```diff
@@ -2,18 +2,19 @@
...
 
 namespace LeaderboardBackend.Test;
 
 [TestFixture]
 internal class Modships
 {
 	private static TestApiFactory Factory = null!;
-	private static HttpClient ApiClient = null!;
+	private static TestApiClient ApiClient = null!;
 	private static JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions
```
[the remark this answers is not in the document]
Comment (RageCage64) [reply LeaderboardBackend.Test/Modships.cs:18]: I hate that style myself, other folks seem to like it though

File: LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs [hunk 3]
```diff
@@ -29,13 +27,12 @@ string password
 			}
 		);
 
-	public static async Task<LoginResponse> Login(
-		HttpClient apiClient,
+	public static async Task<LoginResponse> LoginUser(
+		this TestApiClient apiClient,
```
[the remark this answers is not in the document]
Comment (RageCage64) [reply LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs:31]: This was called `apiClient` before I changed them into extension methods, why the name change?

File: LeaderboardBackend.Test/TestApi/HttpRequestInit.cs [hunk 4]
```diff
@@ -1,6 +1,6 @@
 using System.Net.Http;
 
-namespace LeaderboardBackend.Test.Lib;
+namespace LeaderboardBackend.Test.TestApi;
 
 internal record HttpRequestInit
```
[the remark this answers is not in the document]
Comment (RageCage64) [reply LeaderboardBackend.Test/TestApi/HttpRequestInit.cs:5]: Probably could, it's pretty small and likely won't change.

File: LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs [hunk 5]
```diff
@@ -1,21 +1,19 @@
...
-using System.Text.Json;
+using LeaderboardBackend.Models.Requests;
+using LeaderboardBackend.Test.Lib;
 using System.Threading.Tasks;
 
-namespace LeaderboardBackend.Test.Lib;
+namespace LeaderboardBackend.Test.TestApi.Extensions;
 
-internal static class UserHelpers
+internal static class UserApiExtensions
```
[the remark this answers is not in the document]
Comment (RageCage64) [reply LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs:8]: Yep! To relate it to Go, you can think of
`void Function(this SomeType s)`
As equivalent to Go's version
`func (SomeType s) Function()`

File: LeaderboardBackend.Test/TestApi/TestApiFactory.cs [hunk 6]
```diff
@@ -37,6 +37,12 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
 		});
 	}
 
+	public TestApiClient CreateTestApiClient()
+	{
+		HttpClient client = this.CreateClient();
+		return new TestApiClient(client);
+	}
```
[the remark this answers is not in the document]
Comment (RageCage64) [reply LeaderboardBackend.Test/TestApi/TestApiFactory.cs:44]: I don't think that's going to be doable. Every Factory gets its own special configuration for its HttpClient, so all tests would be fighting over the same TestApiClient instance and trying to put their own HttpClients in, which leaves us in the same place as it was before the PR.

File: LeaderboardBackend/Models/Requests/ParticipationRequests.cs (hunk 1 above)
Comment (RageCage64) [reply LeaderboardBackend/Models/Requests/ParticipationRequests.cs:25]: Yeah probably out of scope, but I can add a FIXME comment

File: LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (zysim) [reply LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs:31]: I figure that there's enough context here to know that `client` is an ApiClient

File: LeaderboardBackend.Test/Modships.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (zysim) [reply LeaderboardBackend.Test/Modships.cs:18]: Up to us I'd say. Let's just keep it then.

File: LeaderboardBackend.Test/TestApi/HttpRequestInit.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (zysim) [reply LeaderboardBackend.Test/TestApi/HttpRequestInit.cs:5]: Yeh let's do that then

File: LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (zysim) [reply LeaderboardBackend.Test/TestApi/Extensions/UserApiExtensions.cs:8]: :o

File: LeaderboardBackend.Test/TestApi/TestApiFactory.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (zysim) [reply LeaderboardBackend.Test/TestApi/TestApiFactory.cs:44]: Ah I see. Well then.

File: LeaderboardBackend/Models/Requests/ParticipationRequests.cs (hunk 1 above)
Comment (zysim) [reply LeaderboardBackend/Models/Requests/ParticipationRequests.cs:25]: I'll do it

**Pull request merged**